gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py

```diff
@@ -0,0 +1,159 @@
+from gbpservice.nfp.configurator.agents import nfp_service as ns
+from gbpservice.nfp.configurator.lib import nfp_service_constants as const
+
+
+class NfpServiceRpcManagerTestCase(base.BaseTestCase):
+    """ Implement test cases for RPC manager methods of nfp service agent.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(NfpServiceRpcManagerTestCase, self).__init__(
+                                                        *args, **kwargs)
+        self.fo = fo.FakeObjects()
+
+    @mock.patch(__name__ + '.fo.FakeObjects.sc')
+    @mock.patch(__name__ + '.fo.FakeObjects.conf')
+    def _get_NfpServiceRpcManager_object(self, conf, sc):
+        """ Retrieves RPC manager object of nfp service agent.
+
+        :param sc: mocked service controller object of process model framework
+        :param conf: mocked OSLO configuration file
+
+        Returns: object of nfp service's RPC manager
+        and service controller.
+
+        """
+
+        agent = ns.ConfigScriptRpcManager(sc, conf)
+        return agent, sc
+
+    def _test_event_creation(self, method):
+        """ Tests event creation and enqueueing for create/delete
+        operation of generic config agent's RPC manager.
+
+        :param method: CREATE_NFP_SERVICE
+
+        Returns: none
+
+        """
+
+        agent, sc = self._get_NfpServiceRpcManager_object()
+        arg_dict = {'context': self.fo.context,
+                    'resource_data': self.fo.kwargs}
+        with mock.patch.object(
+                    sc, 'new_event', return_value='foo') as mock_sc_event, \
+            mock.patch.object(sc, 'post_event') as mock_sc_rpc_event:
+            actual_call = agent.run_nfp_service(self.fo.context,
+                                                self.fo.kwargs)
+
+            expected_cal = mock_sc_event.assert_called_with(
+                                id=method, data=arg_dict, key=None)
+            self.assertEqual(actual_call, expected_cal)
+            mock_sc_rpc_event.assert_called_with('foo')
+
+    def test_nfp_service_rpcmanager(self):
+        """ Implements test case for run_nfp_service method
+        of nfp service RPC manager.
+
+        Returns: none
+
+        """
+
+        self._test_event_creation(const.CREATE_NFP_SERVICE_EVENT)
+
+
+class NfpServiceEventHandlerTestCase(base.BaseTestCase):
+    """ Implements test cases for event handler methods
+    of nfp service agent.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(NfpServiceEventHandlerTestCase, self).__init__(
+                                                        *args, **kwargs)
+        self.fo = fo.FakeObjects()
+        self.context = {'notification_data': {},
+                        'resource': 'interfaces'}
+
+    @mock.patch(__name__ + '.fo.FakeObjects.rpcmgr')
+    @mock.patch(__name__ + '.fo.FakeObjects.drivers')
+    @mock.patch(__name__ + '.fo.FakeObjects.sc')
+    def _get_NfpServiceEventHandler_object(self, sc, drivers, rpcmgr):
```

Comment:
Can we name this function without uppercase chars. Follow the same convention as other function names.